### PR TITLE
Use eslint-plugin-hapi embedded config

### PR DESCRIPTION
**Pull request by kanongil**
Draft changes needed to use the new plugin-based config from https://github.com/hapijs/eslint-plugin-hapi/pull/20. Final PR is dependent on it being published, and its name and version.

### CODE REVIEW

**Comment by devinivy** [comment]
@cjihrig @kanongil are we in a place where eslint-plugin-hapi is ready to be published per https://github.com/hapijs/eslint-plugin-hapi/issues/21?

**Comment by cjihrig** [comment]
I think so. Once https://github.com/hapijs/eslint-plugin-hapi/issues/21 happens, we need to publish a new version of `eslint-plugin-hapi`, but we also will want to deprecate `eslint-config-hapi` and the custom ESLint rules that now live in the plugin.

**Comment by kanongil** [comment]
I have updated the draft PR with the renamed package. To finalise, it just needs an update to the entry in `package.json` with the actual release, after https://github.com/hapijs/eslint-plugin-hapi/pull/23 is merged.

File: package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     ],
     "dependencies": {
         "@hapi/bossy": "5.x.x",
-        "@hapi/eslint-config-hapi": "13.x.x",
+        "@hapi/eslint-plugin": "github:kanongil/eslint-plugin-hapi#rename",
         "@hapi/eslint-plugin-hapi": "4.x.x",
```
Comment (cjihrig) [inline package.json:19]: I published v5.0.0 of the eslint plugin with your changes.

Comment (kanongil) [reply package.json]: Awesome – I have finalized the PR.

File: API.md [hunk 2]
```diff
@@ -583,9 +583,9 @@ As stated at the beginning of the document, `--ignore` parameter is an alias for
 
 **lab** uses a shareable [eslint](http://eslint.org/) config, and a plugin containing several **hapi** specific linting rules. If you want to extend the default linter you must:
```
Comment (Nargonath) [inline API.md:584]: Shouldn't we remove the mention of the shareable config since it is not directly exposed now?

Comment (kanongil) [reply API.md]: I updated the wording so it only reference the plugin (with a config).

**Review by Nargonath — APPROVED** [review]

**Review by cjihrig — APPROVED** [review]

**Pull request merged**